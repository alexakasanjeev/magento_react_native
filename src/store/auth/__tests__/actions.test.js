import {
  signIn,
  signUp,
  resetPassword,
  resetAuthState,
} from '../actions';

describe('auth actions', () => {
  test('signIn()', () => {
    // Setup
    const email = 'sanjeev@gmail.com';
    const password = '123456789';

    // Exercise
    const result = signIn(email, password);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('signUp()', () => {
    // Setup
    const customer = {
      firstname: 'Sanjeev',
      lastname: 'Yadav',
      email: 'sajeev@gmail.com',
    };
    const payload = {
      customer,
      password: '123456789',
    };

    // Exercise
    const result = signUp(payload);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('resetPassword()', () => {
    // Setup
    const email = 'sanjeev@gmail.com';

    // Exercise
    const result = resetPassword(email);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('resetAuthState()', () => {
    // Exercise
    const result = resetAuthState();

    // Verify
    expect(result).toMatchSnapshot();
  });
});
