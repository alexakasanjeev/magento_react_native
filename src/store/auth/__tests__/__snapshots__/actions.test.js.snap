// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth actions resetAuthState() 1`] = `
Object {
  "type": "RESET_AUTH_STATE",
}
`;

exports[`auth actions resetPassword() 1`] = `
Object {
  "payload": Object {
    "email": "sanjeev@gmail.com",
  },
  "type": "MAGENTO_RESET_PASSWORD_REQUEST",
}
`;

exports[`auth actions signIn() 1`] = `
Object {
  "payload": Object {
    "email": "sanjeev@gmail.com",
    "password": "123456789",
  },
  "type": "MAGENTO_SIGN_IN_REQUEST",
}
`;

exports[`auth actions signUp() 1`] = `
Object {
  "payload": Object {
    "customer": Object {
      "email": "sajeev@gmail.com",
      "firstname": "Sanjeev",
      "lastname": "Yadav",
    },
    "password": "123456789",
  },
  "type": "MAGENTO_SIGN_UP_REQUEST",
}
`;
